### PR TITLE
Restore Claude OAuth inference readiness fix

### LINKED ISSUE #93: Claude reviewer readiness must prove OAuth-only non-interactive inference
## Problem

Claude review slots can fail even when `claude auth status` reports a valid Claude Max OAuth login. In the observed session, the plugin's default Claude companion path correctly used subscription/OAuth mode and ignored API-key env vars, but the underlying OAuth-only non-interactive Claude Code inference failed with HTTP 401.

This creates a bad review UX: operators may see Claude as logged in, but every `$claude:claude-adversarial-review` slot fails before inference. We should not silently fall back to API-key auth because the desired policy for this workflow is OAuth-only.

## Hard Evidence From 2026-05-06

Working auth-status probe with API keys removed:

```sh
env -u ANTHROPIC_API_KEY -u CLAUDE_API_KEY claude auth status --text
```

Observed:

```text
Login method: Claude Max account
Organization: seungpyoson@gmail.com's Organization
Email: seungpyoson@gmail.com
```

Failing OAuth-only inference probe:

```sh
env -u ANTHROPIC_API_KEY -u CLAUDE_API_KEY claude -p "reply with exactly: pong" --output-format json
```

Observed:

```json
{"is_error":true,"api_error_status":401,"result":"Failed to authenticate. API Error: 401 Invalid authentication credentials","usage":{"input_tokens":0,"output_tokens":0}}
```

Plugin companion default path reproduced the same failure:

```sh
node plugins/claude/scripts/claude-companion.mjs doctor
```

Observed summary:

```json
{"status":"not_authed","ready":false,"selected_auth_path":"subscription_oauth","auth_policy":"api_key_env_ignored","detail":"Failed to authenticate. API Error: 401 Invalid authentication credentials"}
```

Diagnostic only: `--auth-mode auto` / `--auth-mode api_key` passed by selecting `ANTHROPIC_API_KEY`, proving general Claude access exists, but that path must not count for OAuth-only review approval.

## Desired Behavior

- Claude setup/readiness should include an OAuth-only non-interactive inference probe, not only `claude auth status`.
- The failure should be classified distinctly from generic `claude_error`, e.g. OAuth status present but print/inference token rejected.
- Diagnostics should explicitly state whether API-key env vars were present and ignored, without printing values.
- Review commands should not use API-key fallback unless the user explicitly asks for API-key mode.
- Documentation should explain that `claude auth status` can be a false positive for plugin review readiness.

## Non-Goals

- Do not store or print secrets.
- Do not switch the review path to API-key auth by default.
- Do not perform billing or account mutations.

## Acceptance Criteria

- A failing OAuth-only `claude -p` probe is surfaced by `claude-companion doctor` with a specific actionable status/message.
- Tests cover the case where auth status appears valid but non-interactive OAuth inference returns 401 with zero usage.
- Review evidence clearly marks this as a failed Claude slot rather than a model verdict or approval.

### PR DESCRIPTION
**Pull request by seungpyoson**
## Summary

Restores the Claude OAuth-only readiness fix from PR #94 after the clean-slate reset in PR #102.

This PR reapplies the original PR #94 work onto current `main` and includes current PR #104 review refinements:

- classify Claude subscription/OAuth print-mode HTTP 401 as `oauth_inference_rejected` instead of generic `claude_error`
- make `claude-companion doctor` cross-check `claude auth status --json` when OAuth inference fails, while sanitizing email/org values from output
- use the shared OAuth 401 classifier across preflight, ping, post-spawn JobRecord, and audit-manifest paths
- parse `claude auth status --json` output by balanced JSON object extraction and auth-status semantic filtering so trailing notices and structured prefix notices cannot hide OAuth readiness failures
- isolate OAuth inference ping probes from repository cwd while preserving default PATH lookup, explicit relative paths like `./claude`, and path-like relative binaries like `bin/claude`
- preserve signal/timeout `runCommand()` status as `null` instead of normalizing it to `0`, across Claude/Gemini/Kimi mirrored process helpers
- refactor Sonar-flagged nested/complex review-status, auth-status, JSON-scanning, ping, and JobRecord classification logic without changing behavior
- keep external-review source transmission accurate for preflight and post-spawn failed review slots and restore Grok sync coverage in `sync-external-review.mjs`

Closes #93.

## Context

PR #94 originally implemented this fix and was merged without the requested approval boundary. PR #100 then reverted PR #94 and added two collateral follow-up commits. PR #102 restored the repository tree to the exact pre-PR #94 baseline, removing both PR #94 and PR #100 collateral work from `main`.

This PR is the clean replacement for the #93 fix after that reset. It started from the original PR #94 final tree and now includes only PR #104 review refinements for the same Claude OAuth readiness surface.

## Exact Lineage

- Current base `main`: `a96ed61dec534c29529c2ae2c38c7b6f6fb7b826`
- Current head: `229a43cb0355105d6fb39544cdc2325442a39ea4`
- Original PR #94 base: `7f775882e156843e251560e9b5e81dcf45112e90`
- Original PR #94 head: `e452ea2dfe45dd4bb7b01e8ae0eb03fab79de7b4`
- Recreated commits on this branch:
  - `66a22e0` from original `ebace9fdadf5060612a32d93ff6371d80e7e0486`
  - `8949718` from original `e452ea2dfe45dd4bb7b01e8ae0eb03fab79de7b4`
  - `a308ad2` hardens Claude OAuth preflight failures found during internal review
  - `d4b5c5e` aligns OAuth readiness classification and auth-status JSON parsing found during full path review
  - `773564b` addresses relayed external review blockers from Kimi, GLM, Gemini, and DeepSeek
  - `3fe4ba2` resolves validated Sonar, Gemini Code Assist, and Greptile review findings with systematic refactors and the process timeout-status contract fix
  - `0146753` preserves PATH lookup for default bare `claude` while keeping tmpdir probe isolation and relative-binary support
  - `229a43c` hardens path-like binary resolution, structured auth-status JSON selection, and `cmdPing` control flow after internal adversarial review

## Verification

Run on current head `229a43cb0355105d6fb39544cdc2325442a39ea4`:

- `node --test --test-name-pattern "path-like relative binary|structured prefix notices|default bare claude binary|relative binary" tests/smoke/claude-companion.smoke.test.mjs` -> 6 pass, 0 fail
- `node --test tests/unit/job-record.test.mjs tests/unit/process.test.mjs` -> 74 pass, 0 fail
- `node --test --test-name-pattern "OAuth|audit manifest|doctor|ping" tests/smoke/claude-companion.smoke.test.mjs` -> 27 pass, 0 fail
- `git diff --check` -> pass
- `npm run lint:sync` -> pass
- `npm test` before commit -> 1034 tests, 1028 pass, 0 fail, 6 skipped
- pre-commit `npm test` -> 1034 tests, 1028 pass, 0 fail, 6 skipped

Earlier verification on head `0146753a69e4610ba91319a03d84513487c6254a`:

- `node --test --test-name-pattern "default bare claude binary|relative binary" tests/smoke/claude-companion.smoke.test.mjs` -> 3 pass, 0 fail
- `node --test --test-name-pattern "OAuth|audit manifest|doctor|ping" tests/smoke/claude-companion.smoke.test.mjs` -> 24 pass, 0 fail
- `git diff --check` -> pass
- `npm run lint:sync` -> pass
- `npm test` before commit -> 1031 tests, 1025 pass, 0 fail, 6 skipped
- pre-commit `npm test` -> 1031 tests, 1025 pass, 0 fail, 6 skipped
- GitHub CI on `0146753` -> pass for SonarCloud Code Analysis, lint, test, smoke (claude), smoke (gemini), smoke (kimi), smoke (grok), smoke (api-reviewers)

Earlier verification on head `3fe4ba2bad9e7480b8e188bc52708aae52b5ddd6`:

- `node --test tests/unit/job-record.test.mjs tests/unit/process.test.mjs` -> 74 pass, 0 fail
- `node --test --test-name-pattern "OAuth|audit manifest|doctor|ping" tests/smoke/claude-companion.smoke.test.mjs` -> 22 pass, 0 fail
- `npm run lint:sync` -> pass
- `git diff --check` -> pass
- process helper mirror parity:
  - `cmp -s plugins/claude/scripts/lib/process.mjs plugins/gemini/scripts/lib/process.mjs` -> 0
  - `cmp -s plugins/claude/scripts/lib/process.mjs plugins/kimi/scripts/lib/process.mjs` -> 0
- `npm test` before commit -> 1029 tests, 1023 pass, 0 fail, 6 skipped
- pre-commit `npm test` -> 1029 tests, 1023 pass, 0 fail, 6 skipped
- GitHub CI on `3fe4ba2` -> pass for SonarCloud Code Analysis, lint, test, smoke (claude), smoke (gemini), smoke (kimi), smoke (grok), smoke (api-reviewers)
- Sonar API after `3fe4ba2` analysis: `total: 0`, `issues: []` for `pullRequest=104&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true`

Earlier verification on head `773564b27a16a65f629e62f1fc2cf62f8650be58`:

- `node --test --test-name-pattern "isOAuthInferenceRejected fails closed without auth context|buildJobRecord: OAuth inference 401 is distinct from generic claude_error" tests/unit/job-record.test.mjs` -> 2 pass, 0 fail
- `node --test --test-name-pattern "audit manifest uses canonical claude_error" tests/smoke/claude-companion.smoke.test.mjs` -> 1 pass, 0 fail
- `node --test --test-name-pattern "doctor: OAuth status success but non-interactive inference 401 is classified distinctly|run: OAuth preflight auth-status resolves relative binary from invocation cwd" tests/smoke/claude-companion.smoke.test.mjs` -> 2 pass, 0 fail
- `node --test tests/unit/job-record.test.mjs` -> 65 pass, 0 fail
- `node --test --test-name-pattern "OAuth|audit manifest|doctor|ping" tests/smoke/claude-companion.smoke.test.mjs` -> 22 pass, 0 fail
- `git diff --check` -> pass
- `npm run lint:sync` -> pass
- `npm test` first run -> 1029 tests, 1022 pass, 1 fail, 6 skipped; failed `sync-browser-session preserves grok2api_timeout during mid-import hangs` with `grok2api_unreachable` instead of `grok2api_timeout`
- Rerun of exact failed Grok smoke -> 1 pass, 0 fail
- `npm test` second run -> 1029 tests, 1023 pass, 0 fail, 6 skipped
- pre-commit `npm test` -> 1029 tests, 1023 pass, 0 fail, 6 skipped

Earlier verification on head `d4b5c5eb30df8092eac6651f43bc08cbc32b1aaf`:

- `git diff --check` -> pass
- `npm run lint:sync` -> pass
- `node --test --test-name-pattern "doctor: OAuth status JSON tolerates trailing notices with braces|run: post-spawn OAuth inference 401 is mirrored into audit manifest" tests/smoke/claude-companion.smoke.test.mjs` -> 2 pass, 0 fail
- `node --test --test-name-pattern "OAuth" tests/smoke/claude-companion.smoke.test.mjs` -> 7 pass, 0 fail
- `node --test tests/smoke/claude-companion.smoke.test.mjs` -> 68 pass, 0 fail
- `node --test tests/unit/job-record.test.mjs tests/unit/process.test.mjs` -> 73 pass, 0 fail
- `npm test` -> 1027 tests, 1021 pass, 0 fail, 6 skipped
- pre-commit `npm test` -> 1027 tests, 1021 pass, 0 fail, 6 skipped

Initial recreated-PR verification before refinements:

- `git diff --name-only e452ea2dfe45dd4bb7b01e8ae0eb03fab79de7b4 HEAD` -> empty output at recreated head `89497185fbaba0a359c4c08249aa86791b1bcc74`
- `env -u ANTHROPIC_API_KEY -u CLAUDE_API_KEY claude auth status --json | node -e ...` confirmed sanitized auth-status field shape during original PR #94 verification

## Comment Resolution

- Greptile `parseJsonObjectOutput` trailing-brace comment: fixed and replied with evidence at comment IDs `3195303626` and `3195850149`; thread resolved by Greptile.
- Gemini Code Assist structured OAuth classifier comment: fixed by exporting/reusing `isOAuthInferenceRejected()` and canonical `classifyExecution()`; replied with evidence at comment IDs `3195315726` and `3195852203`; thread resolved.
- Greptile timeout assertion comment: fixed by preserving raw `spawnSync` timeout/signal `status: null` in Claude/Gemini/Kimi process helpers and updating `tests/unit/process.test.mjs`; replied with evidence at comment IDs `3195371884` and `3195851208`; thread resolved.
- SonarCloud new issues: API result after `3fe4ba2` was `total: 0`, `issues: []`; evidence posted in top-level comment `4388649159`.
- Relayed Kimi/DeepSeek blocker on `3fe4ba2` for default bare `claude` PATH lookup: fixed in `0146753` by resolving only explicit relative binaries and adding PATH-based ping and run-preflight smoke tests.
- Internal adversarial review findings after `0146753`: fixed in `229a43c` by resolving path-like relative binaries such as `bin/claude`, filtering balanced JSON candidates to auth-status-shaped objects, making `cmdPing` helper exits explicit with `return`, using optional chaining on the parsed-missing fallback, and passing a named auth-selection classifier context to `isOAuthInferenceRejected()`.

## Relayed External Review Status

Reviews relayed for stale head `3fe4ba2bad9e7480b8e188bc52708aae52b5ddd6`:

- GLM: APPROVE. No blockers.
- Claude: APPROVE. No blockers.
- Gemini: APPROVE. No blockers.
- Kimi: DO NOT APPROVE. Blocking finding: unconditional `resolvePath()` broke PATH lookup for default bare `claude`, breaking `ping`/`doctor` and silently skipping OAuth run preflight. Addressed in `0146753`.
- DeepSeek: DO NOT APPROVE. Same blocking finding: PATH binary resolution regression. Addressed in `0146753`.

Reviews relayed for stale head `d4b5c5eb30df8092eac6651f43bc08cbc32b1aaf`:

- Kimi: DO NOT APPROVE. Blocking finding: `isOAuthInferenceRejected()` failed open without invocation/auth context. Addressed in `773564b` by returning `false` when auth context is absent and adding a unit test.
- GLM: DO NOT APPROVE. Blocking finding: `review_metadata.audit_manifest.error_code` used parsed reason instead of canonical `classifyExecution()` output for non-OAuth failures. Addressed in `773564b` by exporting/reusing `classifyExecution()` and adding a smoke test for canonical `claude_error`.
- Gemini: DO NOT APPROVE. Blocking finding: OAuth inference ping probes ran from repo/caller cwd and could trigger large-repo startup/indexing timeout. Addressed in `773564b` by running inference probes in `tmpdir()` with relative binary resolution preserved and adding cwd-isolation smoke assertions.
- DeepSeek: DO NOT APPROVE. Blocking finding: OAuth diagnostic text said source was not sent even for post-spawn OAuth 401 where source transmission was `sent`. Addressed in `773564b` by removing the source-transmission claim from the generic OAuth diagnostic and updating the unit assertion.
- Claude: APPROVE on stale head `d4b5c5e`; also identified the null-auth-context footgun as non-blocking. Superseded by `773564b`.

No latest-head external/adversarial review has been run on current head `229a43cb0355105d6fb39544cdc2325442a39ea4`.

### CODE REVIEW

File: plugins/claude/scripts/claude-companion.mjs [hunk 1]
```diff
@@ -1237,6 +1300,63 @@ function pingErrorFields() {
...
 }
 
+function oauthInferenceRejectedFields() {
+  return {
+    ready: false,
+    summary: "Claude Code OAuth login is present, but OAuth non-interactive inference is rejected.",
+    next_action: "Refresh Claude OAuth in a normal terminal with `claude auth login`, then verify OAuth-only `claude -p` inference works.",
+  };
+}
+
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply plugins/claude/scripts/claude-companion.mjs:1346]: Fixed at head `d4b5c5eb30df8092eac6651f43bc08cbc32b1aaf`.

Evidence:
- `plugins/claude/scripts/claude-companion.mjs` now uses balanced JSON-object extraction instead of first `{` to last `}` fallback.
- Added smoke coverage: `doctor: OAuth status JSON tolerates trailing notices with braces`.
- Red failure before fix: `actual 'not_authed'`, expected `oauth_inference_rejected`.
- Green verification after fix: `node --test --test-name-pattern "OAuth" tests/smoke/claude-companion.smoke.test.mjs` -> `7 pass, 0 fail`.
- Full verification after fix: `npm test` -> `1027 tests, 1021 pass, 0 fail, 6 skipped`.

File: plugins/claude/scripts/claude-companion.mjs [hunk 2]
```diff
@@ -843,6 +875,37 @@ async function spawnClaudeOrExit(invocation, profile, prompt, executionScope, mu
...
+      allowedApiKeyEnv: authSelection.allowed_env_credentials,
+    });
+  } catch {
+    return null;
+  }
+  if (execution.parsed?.ok === true) return null;
+  const detail = pingFailureDetail(execution);
+  if (!PING_AUTH_RE.test(detail)) return null;
+  const oauthStatus = safeClaudeOAuthStatus(invocation.binary, authSelection);
+  if (oauthStatus?.logged_in !== true || !/401|invalid authentication credentials/i.test(detail)) return null;
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply plugins/claude/scripts/claude-companion.mjs]: Addressed at head `d4b5c5eb30df8092eac6651f43bc08cbc32b1aaf`.

Evidence:
- `isOAuthInferenceRejected()` is now exported from `plugins/claude/scripts/lib/job-record.mjs` and reused by `plugins/claude/scripts/claude-companion.mjs` in `claudeOAuthInferencePreflight()`, `cmdPing()`, and `reviewAuditManifest()`.
- This removed the duplicated OAuth 401 string/status classification drift between the companion path and JobRecord path.
- Added smoke coverage for the missing post-spawn path: `run: post-spawn OAuth inference 401 is mirrored into audit manifest`.
- Red failure before fix: `record.error_code` was `oauth_inference_rejected`, but `record.review_metadata.audit_manifest.error_code` was `null`.
- Green verification after fix: `node --test --test-name-pattern "OAuth" tests/smoke/claude-companion.smoke.test.mjs` -> `7 pass, 0 fail`.
- Full verification after fix: `npm test` -> `1027 tests, 1021 pass, 0 fail, 6 skipped`.

File: tests/unit/process.test.mjs [hunk 3]
```diff
@@ -25,6 +25,16 @@ test("runCommand: captures stdout, stderr, status, and command metadata", () =>
...
 });
 
+test("runCommand: supports bounded spawnSync timeouts", () => {
+  const result = runCommand(process.execPath, [
+    "-e",
+    "setTimeout(() => {}, 1000)",
+  ], { timeout: 50 });
+
+  assert.equal(result.status, 0);
+  assert.equal(result.error?.code, "ETIMEDOUT");
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply tests/unit/process.test.mjs]: Verified as a false positive for the current implementation; no code change made for this comment.

Evidence at head `d4b5c5eb30df8092eac6651f43bc08cbc32b1aaf`:
- `plugins/claude/scripts/lib/process.mjs` intentionally normalizes `result.status ?? 0`, so the wrapper contract returns `0` when Node's raw `spawnSync` status is `null` on timeout.
- Focused verification: `node --test tests/unit/job-record.test.mjs tests/unit/process.test.mjs` -> `73 pass, 0 fail`.
- Full verification: `npm test` -> `1027 tests, 1021 pass, 0 fail, 6 skipped`.

The assertion is testing the wrapper contract, not raw Node `spawnSync` output.

File: plugins/claude/scripts/claude-companion.mjs (hunk 1 above)
[the remark this answers is not in the document]
Comment (seungpyoson) [reply plugins/claude/scripts/claude-companion.mjs:1346]: Addressed at head `3fe4ba2bad9e7480b8e188bc52708aae52b5ddd6`.

Evidence:
- The balanced JSON scanner in `plugins/claude/scripts/claude-companion.mjs` is now split into `parseFirstBalancedJsonObject()`, `parseBalancedJsonCandidate()`, `updateJsonScanState()`, `updateJsonStringState()`, and `parseJsonSlice()` so the parser no longer uses first-brace/last-brace extraction and the scanner logic is bounded.
- Existing coverage for the original edge case remains green: `doctor: OAuth status JSON tolerates trailing notices with braces`.
- Focused verification: `node --test --test-name-pattern "OAuth|audit manifest|doctor|ping" tests/smoke/claude-companion.smoke.test.mjs` -> `22 pass, 0 fail`.
- Full verification before commit: `npm test` -> `1029 tests, 1023 pass, 0 fail, 6 skipped`.
- Pre-commit full verification: `npm test` -> `1029 tests, 1023 pass, 0 fail, 6 skipped`.

File: tests/unit/process.test.mjs (hunk 3 above)
[the remark this answers is not in the document]
Comment (seungpyoson) [reply tests/unit/process.test.mjs]: Addressed at head `3fe4ba2bad9e7480b8e188bc52708aae52b5ddd6`.

Evidence:
- `runCommand()` now preserves Node `spawnSync` timeout/signal status as `null` instead of normalizing it to `0` in all mirrored process helpers:
  - `plugins/claude/scripts/lib/process.mjs`
  - `plugins/gemini/scripts/lib/process.mjs`
  - `plugins/kimi/scripts/lib/process.mjs`
- `tests/unit/process.test.mjs` now asserts `result.status === null` and `result.error.code === "ETIMEDOUT"` for bounded timeouts.
- Mirror verification: `cmp -s plugins/claude/scripts/lib/process.mjs plugins/gemini/scripts/lib/process.mjs` -> `0`; `cmp -s plugins/claude/scripts/lib/process.mjs plugins/kimi/scripts/lib/process.mjs` -> `0`.
- Focused verification: `node --test tests/unit/job-record.test.mjs tests/unit/process.test.mjs` -> `74 pass, 0 fail`.
- Full verification before commit and pre-commit both passed: `1029 tests, 1023 pass, 0 fail, 6 skipped`.

File: plugins/claude/scripts/claude-companion.mjs (hunk 2 above)
[the remark this answers is not in the document]
Comment (seungpyoson) [reply plugins/claude/scripts/claude-companion.mjs]: Addressed and kept addressed at head `3fe4ba2bad9e7480b8e188bc52708aae52b5ddd6`.

Evidence:
- OAuth 401 classification remains centralized through `isOAuthInferenceRejected()` and canonical JobRecord classification through exported `classifyExecution()`.
- `reviewAuditManifest()` now delegates status derivation to `reviewAuditStatus()` and error-code derivation to `classifyExecution()`, removing local classifier drift and the nested status ternary.
- Focused verification: `node --test tests/unit/job-record.test.mjs tests/unit/process.test.mjs` -> `74 pass, 0 fail`.
- OAuth/ping/doctor smoke verification: `node --test --test-name-pattern "OAuth|audit manifest|doctor|ping" tests/smoke/claude-companion.smoke.test.mjs` -> `22 pass, 0 fail`.
- Full verification before commit and pre-commit both passed: `1029 tests, 1023 pass, 0 fail, 6 skipped`.

**Comment by seungpyoson** [comment]
Addressed Sonar findings at head `3fe4ba2bad9e7480b8e188bc52708aae52b5ddd6`.

Evidence:
- Previous Sonar API result on head `773564b27a16a65f629e62f1fc2cf62f8650be58`: `total: 5` open/confirmed new issues.
- Current Sonar API result on head `3fe4ba2bad9e7480b8e188bc52708aae52b5ddd6`: `total: 0`, `issues: []` for `pullRequest=104&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true`.
- SonarCloud Code Analysis check: pass.
- GitHub checks on current head: lint, test, smoke (claude), smoke (gemini), smoke (kimi), smoke (grok), smoke (api-reviewers): all pass.
- Local verification before commit: `npm test` -> `1029 tests, 1023 pass, 0 fail, 6 skipped`.
- Pre-commit verification: `npm test` -> `1029 tests, 1023 pass, 0 fail, 6 skipped`.

**Comment by seungpyoson** [comment]
Addressed valid internal adversarial review findings at head `229a43cb0355105d6fb39544cdc2325442a39ea4`.

Evidence:
- `--binary bin/claude` blocker: fixed by treating path-like relative binaries containing `/` or `\` as cwd-relative before tmpdir probe spawn. Added smoke coverage for both `ping --binary bin/claude` and run preflight `--binary bin/claude`.
- Greptile top-level issue 1: `execution.parsed.raw` fallback hardened to `execution.parsed?.raw`.
- Greptile top-level issue 2: `cmdPing()` now returns immediately after helper calls that terminate via `process.exit()`, making the control flow explicit.
- Greptile top-level issue 3: `printPingExecutionFailure()` now passes an explicit `authSelectionClassifierContext()` into `isOAuthInferenceRejected()`, and the classifier parameter is named `authContext` instead of `invocation`.
- Structured prefix JSON gap: auth-status parsing now accepts only auth-status-shaped JSON objects, so a prefix object such as `{ "event": "refreshing" }` is skipped before parsing the real `{ "loggedIn": true, ... }` object.

Verification on current head:
- `node --test --test-name-pattern "path-like relative binary|structured prefix notices|default bare claude binary|relative binary" tests/smoke/claude-companion.smoke.test.mjs` -> 6 pass, 0 fail
- `node --test tests/unit/job-record.test.mjs tests/unit/process.test.mjs` -> 74 pass, 0 fail
- `node --test --test-name-pattern "OAuth|audit manifest|doctor|ping" tests/smoke/claude-companion.smoke.test.mjs` -> 27 pass, 0 fail
- `git diff --check` -> pass
- `npm run lint:sync` -> pass
- `npm test` before commit -> 1034 tests, 1028 pass, 0 fail, 6 skipped
- pre-commit `npm test` -> 1034 tests, 1028 pass, 0 fail, 6 skipped

**Comment by seungpyoson** [comment]
## PR #104 current status

Head: `8df8c0d87cfed20b33ef004b4977f3ecacbb9228`
Branch: `codex/restore-claude-oauth-readiness`
Base: `a96ed61dec534c29529c2ae2c38c7b6f6fb7b826`

### Latest follow-up

Addressed the latest review-output residuals with a bounded hardening commit:

- Normalized non-`Error` preflight throws before persisting `errorMessage`.
- Documented why `claude auth status --json` keeps invocation cwd while the inference probe runs in `tmpdir()`.
- Hardened auth-status JSON candidate selection so weak prefix objects such as `{ "loggedIn": false }` do not mask the real status object that follows.
- Added smoke/unit coverage for:
  - weak `loggedIn: false` prefix before real auth status,
  - auth-status nonzero exit,
  - auth-status signal termination,
  - auth-status ENOENT after ping,
  - parent-relative `../bin/claude` binary resolution,
  - background OAuth preflight rejection with `not_sent` and `failed_review_slot: false`,
  - cancellation during OAuth preflight before target launch,
  - `api_key_env_missing` 401 staying generic `claude_error`.

### Local verification on `8df8c0d`

- `node --test --test-name-pattern "loggedIn prefix|nonzero exit|killed by signal|missing binary after ping|parent-relative binary|background: OAuth preflight|cancel during OAuth preflight|missing API-key path" tests/smoke/claude-companion.smoke.test.mjs tests/unit/job-record.test.mjs` -> 8 pass, 0 fail
- `node --test --test-name-pattern "OAuth|oauth|preflight|isOAuthInferenceRejected|API-key 401|missing API-key path|binary" tests/smoke/claude-companion.smoke.test.mjs tests/unit/job-record.test.mjs tests/unit/process.test.mjs` -> 31 pass, 0 fail
- `node --test tests/unit/job-record.test.mjs tests/unit/process.test.mjs tests/unit/companion-common.test.mjs` -> 91 pass, 0 fail
- `npm run lint:sync` -> pass
- `node --test tests/smoke/claude-companion.smoke.test.mjs` -> 83 pass, 0 fail
- `git diff --check` -> pass
- `npm test` -> 1045 tests, 1039 pass, 0 fail, 6 skipped
- pre-commit `npm test` -> 1045 tests, 1039 pass, 0 fail, 6 skipped

### GitHub CI on `8df8c0d`

All reported checks passed:

- SonarCloud Code Analysis
- lint
- test
- smoke (api-reviewers)
- smoke (claude)
- smoke (gemini)
- smoke (grok)
- smoke (kimi)

### External review status

The externally relayed approvals were for previous head `b61ec38dee615e2bb47af002037ed549e4e4e6dd`. This follow-up commit intentionally addresses their non-blocking residuals, so those approvals are now stale with respect to the exact current head `8df8c0d`.
